### PR TITLE
feat: Add PI30M054 with support for QBMS

### PR DESCRIPTION
**Pull request by BoboTiG**
```
$ mpp-solar -P PI30M054 -c QBMS
Command: QBMS - Read lithium battery information
--------------------------------------------------------------------------------
Parameter                               Value          	Unit
battery_connect_status                  Connected
battery_capacity_from_bms               38             	%
battery_force_charging                  No
battery_stop_discharge_flag             Enable discharge
battery_stop_charge_flag                Enable charge
battery_bulk_charging_voltage_from_bms  53.2           	V
battery_float_charging_voltage_from_bms 53.2           	V
battery_cut_off_voltage_from_bms        45.0           	V
battery_max_charging_current            120            	A
battery_max_discharge_current           240            	A
--------------------------------------------------------------------------------
```

I copied the whole code from #402, thank you @lugamodder :)

### CODE REVIEW

File: mppsolar/protocols/pi30m054.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        "help": " -- queries the value of various metrics from the battery",
+        "type": "QUERY",
+        "crctype": "chk",
+        "response": [
+            ["keyed", "Battery connect status", {"0": "Connected", "1": "Disconnected"}],
+            ["int", "Battery capacity from BMS", "%"],
+            ["keyed", "Battery force charging", {"0": "No", "1": "Yes"}],
+            ["keyed", "Battery stop discharge flag", {"0": "Enable discharge", "1": "Disable discharge"}],
+            ["keyed", "Battery stop charge flag", {"0": "Enable charge", "1": "Disable charge"}],
+            ["int", "Battery bulk charging voltage from BMS", "0.1 V"],
```
Comment (BoboTiG) [inline mppsolar/protocols/pi30m054.py:20]: @jblance I tried what you suggested on #402 with `int:r*10`, vut if ends on:
```python
  File "mppsolar/outputs/screen.py", line 60, in output
    key = key.replace(" ", "_")
          ^^^^^^^^^^^
AttributeError: 'int' object has no attribute 'replace'
```

for the sake of completness, I tried the `json` output and got the same error. Do you have an idea how to workaround that?

Comment (jblance) [reply mppsolar/protocols/pi30m054.py:20]: it needs to be the INDEXED format 
```python
"QBMS": {
        "name": "QBMS",
        "description": "Read lithium battery information",
        "help": " -- queries the value of various metrics from the battery",
        "type": "QUERY",
        "crctype": "chk",
        "response_type": "INDEXED",
        "response": [
            [0, "Battery connect status", "str_keyed", {"0": "Connected", "1": "Disconnected"}],
            [1, "Battery capacity from BMS", "int", "%"],
            [2, "Battery force charging", "str_keyed", {"0": "No", "1": "Yes"}],
            [3, "Battery stop discharge flag", "str_keyed", {"0": "Enable discharge", "1": "Disable discharge"}],
            [4, "Battery stop charge flag", "str_keyed", {"0": "Enable charge", "1": "Disable charge"}],
            [5, "Battery bulk charging voltage from BMS", "int:r/10", "V"],
            [6, "Battery float charging voltage from BMS", "int:r/10", "V"],
            [7, "Battery cut off voltage from BMS", "int:r/10", "V"],
            [8, "Battery max charging current", "float", "A"],
            [9, "Battery max discharge current", "float", "A"]],
        "test_responses": [
            b"(0 086 0 0 0 532 532 450 0120 0240\x7fU\r",
        ],
    },
```

Comment (BoboTiG) [reply mppsolar/protocols/pi30m054.py:20]: Awesome, I tried several things but not that. Thanks a lot!

**Comment by BoboTiG** [comment]
Changes applied, it works like a charm 👍🏻 

Would you mind updating https://github.com/jblance/mpp-solar/wiki/Protocol-Recommendations with the new protocol 🙏🏻 ?

**Comment by jblance** [comment]
> Changes applied, it works like a charm 👍🏻
> 
> Would you mind updating https://github.com/jblance/mpp-solar/wiki/Protocol-Recommendations with the new protocol after the merge 🙏🏻 ?

As a test, can you try editing the wiki - its supposed to allow 'contributors' to edit (and you are one...)

**Comment by jblance** [comment]
> Still can't edit wiki pages.

bummer, ok updated (hopefully how you intended)

**Comment by BoboTiG** [comment]
> > Still can't edit wiki pages.
> 
> bummer, ok updated (hopefully how you intended)

Perfect, thanks 💪🏻

**Pull request merged**